### PR TITLE
fix: fixed diff test issue 11952

### LINKED ISSUE #11952: (2/19 - 18:00:00) Mismatched values, expected 'ACCOUNT_DELETED', got 'CONTRACT_DELETED'
### Description

Transaction in HashScan [here](https://hashscan.io/mainnet/transaction/1708366312.182137003).

A crypto transfer transaction body:

```
transactionID {
  transactionValidStart {
    seconds: 1708366301
    nanos: 92722239
  }
  accountID {
    accountNum: 3288044
  }
}
nodeAccountID {
  accountNum: 23
}
transactionFee: 100000000
transactionValidDuration {
  seconds: 120
}
memo: "we tryin\' to spam so much that we get featured on the HBAR bull - GRELF Mafia"
cryptoTransfer {
  transfers {
  }
  tokenTransfers {
    token {
      tokenNum: 1159074
    }
    transfers {
      accountID {
        accountNum: 2957496
      }
      amount: 10000
    }
    transfers {
      accountID {
        accountNum: 3288044
      }
      amount: -10000
    }
  }
}
```
Got:
---- CryptoTransfer DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched values, expected 'ACCOUNT_DELETED', got 'CONTRACT_DELETED' - Matching field 'status' Matching field 'receipt'  ==> expected: <ACCOUNT_DELETED> but was: <CONTRACT_DELETED>

### Steps to reproduce

Run 15 mins event from round 162952161

### Additional context

_No response_

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn

### Operating system

None

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

**Related issue(s)**:

Fixes #11952 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by povolev15 — APPROVED** [review]
LGTM, thanks Iris

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenHandlerHelper.java [hunk 1]
```diff
@@ -152,7 +151,7 @@ public static Account getIfUsable(
         validateTrue(acct != null, errorIfNotUsable);
         final var isContract = acct.smartContract();
 
-        validateFalse(acct.deleted(), isContract ? CONTRACT_DELETED : errorOnAccountDeleted);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenHandlerHelper.java:155]: Wouldn't smart contract service reach this code ?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenHandlerHelper.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenHandlerHelper.java:155]: To match what mono's return.

**Pull request merged**